File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    int temp = 1;
    int start = 1;
    int cnt = 0;
    while(temp <= n){
        cnt += temp;
        temp += 1;
        
        if (cnt == n) {
            answer = answer + 1;
            cnt = 0;
            start = start + 1;
            temp = start;
        } 
            
        if (cnt > n) {
            cnt = 0;
            start += 1;
            temp = start;
        }
            
    }
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 1
    temp = 0
    start = 1
    add = 1
    while start != n:
        if temp == n:
            answer += 1
            start += 1
            temp = 0
            add = start
        if temp > n:
            start += 1
            temp = 0
            add = start
        else:
            temp += add
            add += 1
        
        
    return answer